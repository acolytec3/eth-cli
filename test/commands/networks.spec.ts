import NetworksCommand from '../../src/commands/networks'

describe('networks', () => {
  let stdoutResult: any

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString())))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'networks --help' and throw an error.`, async () => {
    await expect(NetworksCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'networks' and match snapshot.`, async () => {
    await NetworksCommand.run([])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'networks --json' and match snapshot.`, async () => {
    await NetworksCommand.run(['--json'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'networks --table' and match snapshot.`, async () => {
    await NetworksCommand.run(['--table'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:urls --bar' and throw an error.`, async () => {
    await expect(NetworksCommand.run(['--bar'])).rejects.toThrow('Unexpected argument: --bar')
  })
})
