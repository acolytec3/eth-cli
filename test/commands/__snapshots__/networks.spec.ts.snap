// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`networks Should run 'networks --json' and match snapshot. 1`] = `
Array [
  "{
  \\"kovan\\": {
    \\"url\\": \\"https://kovan.infura.io\\",
    \\"id\\": 42,
    \\"label\\": \\"Kovan\\"
  },
  \\"mainnet\\": {
    \\"url\\": \\"https://mainnet.infura.io\\",
    \\"id\\": 1,
    \\"label\\": \\"Mainnet\\"
  },
  \\"poa\\": {
    \\"url\\": \\"https://core.poa.network\\",
    \\"id\\": 99,
    \\"label\\": \\"POA\\"
  },
  \\"rinkeby\\": {
    \\"url\\": \\"https://rinkeby.infura.io\\",
    \\"id\\": 4,
    \\"label\\": \\"Rinkeby\\"
  },
  \\"ropsten\\": {
    \\"url\\": \\"https://ropsten.infura.io\\",
    \\"id\\": 3,
    \\"label\\": \\"Ropsten\\"
  },
  \\"rsk\\": {
    \\"url\\": \\"https://public-node.rsk.co\\",
    \\"id\\": 30,
    \\"label\\": \\"RSK\\"
  },
  \\"rsk-testnet\\": {
    \\"url\\": \\"https://public-node.testnet.rsk.co\\",
    \\"id\\": 31,
    \\"label\\": \\"RSK testnet\\"
  },
  \\"sokol\\": {
    \\"url\\": \\"https://sokol.poa.network\\",
    \\"id\\": 77,
    \\"label\\": \\"Sokol\\"
  },
  \\"xdai\\": {
    \\"url\\": \\"https://dai.poa.network\\",
    \\"id\\": 100,
    \\"label\\": \\"xDAI\\"
  }
}
",
]
`;

exports[`networks Should run 'networks --table' and match snapshot. 1`] = `
Array [
  "Id     Name        Url                                
",
  "1      Mainnet     https://mainnet.infura.io          
",
  "3      Ropsten     https://ropsten.infura.io          
",
  "4      Rinkeby     https://rinkeby.infura.io          
",
  "30     RSK         https://public-node.rsk.co         
",
  "31     RSK testnet https://public-node.testnet.rsk.co 
",
  "42     Kovan       https://kovan.infura.io            
",
  "77     Sokol       https://sokol.poa.network          
",
  "99     POA         https://core.poa.network           
",
  "100    xDAI        https://dai.poa.network            
",
]
`;

exports[`networks Should run 'networks' and match snapshot. 1`] = `
Array [
  "{
  \\"kovan\\": {
    \\"url\\": \\"https://kovan.infura.io\\",
    \\"id\\": 42,
    \\"label\\": \\"Kovan\\"
  },
  \\"mainnet\\": {
    \\"url\\": \\"https://mainnet.infura.io\\",
    \\"id\\": 1,
    \\"label\\": \\"Mainnet\\"
  },
  \\"poa\\": {
    \\"url\\": \\"https://core.poa.network\\",
    \\"id\\": 99,
    \\"label\\": \\"POA\\"
  },
  \\"rinkeby\\": {
    \\"url\\": \\"https://rinkeby.infura.io\\",
    \\"id\\": 4,
    \\"label\\": \\"Rinkeby\\"
  },
  \\"ropsten\\": {
    \\"url\\": \\"https://ropsten.infura.io\\",
    \\"id\\": 3,
    \\"label\\": \\"Ropsten\\"
  },
  \\"rsk\\": {
    \\"url\\": \\"https://public-node.rsk.co\\",
    \\"id\\": 30,
    \\"label\\": \\"RSK\\"
  },
  \\"rsk-testnet\\": {
    \\"url\\": \\"https://public-node.testnet.rsk.co\\",
    \\"id\\": 31,
    \\"label\\": \\"RSK testnet\\"
  },
  \\"sokol\\": {
    \\"url\\": \\"https://sokol.poa.network\\",
    \\"id\\": 77,
    \\"label\\": \\"Sokol\\"
  },
  \\"xdai\\": {
    \\"url\\": \\"https://dai.poa.network\\",
    \\"id\\": 100,
    \\"label\\": \\"xDAI\\"
  }
}
",
]
`;
